### PR TITLE
Bugfix account operations

### PR DESCRIPTION
**Pull request by BenjamenMeyer**
Update SwiftMock for support the HEAD and POST operations at the Account level.
These calls are inseparable without ability to access the underlying model/framework to validate the storage details as one adds data and the other retrieves it.

POST allows the ability to add custom metadata on the account.
HEAD allows retrieval of the custom metadata as well as counts of total containers, total objects (in all containers), and total bytes used (by all objects in all containers).

### CODE REVIEW

File: mimic/test/test_swift.py [hunk 1]
```diff
@@ -243,6 +289,98 @@ def test_openstack_ids(self):
...
+                b"X-Account-Meta-MockTest": [b"HelloWorld"],
+                b"x-account-meta-field": [b"WorldWide"]
+            }
+
+        post_response, post_content = self.post_account(headers=headers)
+
+        # validate via a HEAD operation
+        head_response, head_content = self.head_account()
+
+        if headers is not None:
```
Comment (glyph) [inline mimic/test/test_swift.py]: I'll repeat my same uneasiness with control structures in tests from previous reviews, but again, not blocking feedback, just something to think about :)

File: mimic/test/test_swift.py [hunk 2]
```diff
@@ -243,6 +289,98 @@ def test_openstack_ids(self):
...
+                b"X-Account-Bytes-Used")[0],
+            "{0}".format(total_bytes).encode("utf-8"))
+
+    @ddt.data(
+        True,
+        False
+    )
+    def test_post_account_metadata(self, post_metadata):
+        """
+        POSTing an account correctly saves metadata to the account.
```
Comment (glyph) [inline mimic/test/test_swift.py]: "correctly" is another one of those non-words from https://jml.io/pages/test-docstrings.html that we should avoid.

File: mimic/test/test_swift.py [hunk 3]
```diff
@@ -243,6 +289,98 @@ def test_openstack_ids(self):
...
+
+    @ddt.data(
+        (0, 0),
+        (1, 0),
+        (2, 5),
+    )
+    @ddt.unpack
+    def test_head_account(self, container_count, object_count):
+        """
+        HEADing an account correctly reports the container count,
```
Comment (glyph) [inline mimic/test/test_swift.py]: https://jml.io/pages/test-docstrings.html again :)

**Comment by glyph** [comment]
If you could do a little bit of post-hoc cleanup based on the inline feedback that would be great but this looks good to merge to me.

**Comment by BenjamenMeyer** [comment]
@glyph thanks...will address with follow up PR.

**Pull request merged**